alt-tab-macos/logic/Screen.swift

```diff
@@ -14,9 +14,23 @@ class Screen {
 
     static func updateThumbnailMaxSize() -> Void {
         if Preferences.thumbnailMaxWidth == 200 && Preferences.thumbnailMaxHeight == 200 {
-            let main = NSScreen.main!.frame
+            let screenFrame = getPreferredScreen().frame
             Preferences.thumbnailMaxWidth = (NSScreen.main!.frame.size.width * Preferences.maxScreenUsage! - Preferences.windowPadding * 2) / Preferences.maxThumbnailsPerRow! - Preferences.interItemPadding
-            Preferences.thumbnailMaxHeight = Preferences.thumbnailMaxWidth * (main.height / main.width)
+            Preferences.thumbnailMaxHeight = Preferences.thumbnailMaxWidth * (screenFrame.height / screenFrame.width)
         }
     }
+
+    static func getPreferredScreen() -> NSScreen {
+        switch Preferences.showOnScreen! {
+        case .MOUSE:
+            return getScreenWithMouse() ?? NSScreen.main!; // .main as fail-safe
+        default: // no case for .MAIN since its currently the same as default case
```

Comment:
Yeah thats what happened here.

It does not allow a DRY combined case/default. But now I worked on it a little bit the IDE said rightfully the default case is never called (since we said Preferences.showOnScreen! is always filled).

We could also throw here incase NSScreen.main is empty but then we will need to handle it somewhere and decide which methods in the call hierarchy rethrow the exception.

I gonna commit a proposal. Tell me at re-viewing the changes the way you would like to go here (or you can also commit -> you have as a maintainer write permissions to this feature branch).

Reply:
I think the current code is fine. I was curious about not listing the 2 cases directly, but yeah I guess swift compiler isn't smart enough. I remember doing lots of pattern matching in Scala a few years ago and I remember it being always able to check if it was exhaustive or not